server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -98,6 +98,12 @@ public boolean isApplicationEligible() {
     return getAllActiveBlocks().stream().allMatch(block -> isBlockEligible(block.getId()));
   }
 
+  @Override
+  public boolean isInProgressApplicationNotEligible() {
+    return getSummaryData().stream()
+        .anyMatch(question -> !question.isEligible() && question.isAnswered());
```

Comment:
question (blocking): why does this use the summary data whereas `isApplicationEligible` uses the active blocks? Could this not also use the active blocks?

Reply:
I originally went with active blocks, but with that it won't show the not eligible message toast if the applicant has two questions in a block and one of them is answered and not eligible and the other hasn't been answered. It would only show the not eligible message if all of the questions have been answered and any are not eligible. The reason we have to check for isAnswered is because of #4124. By using summary data we can check each question on its own, so we don't have to worry about having some questions on the block answered and some unanswered. Does that make sense? Is there a way of doing this I might be missing?

Reply:
That's a good point, thanks! Updated.